include/pybind11/cast.h

```diff
@@ -772,20 +772,42 @@ using cast_op_type =
         typename std::add_pointer<intrinsic_t<T>>::type,
         typename std::add_lvalue_reference<intrinsic_t<T>>::type>;
 
+template <typename T> using cast_op_type_for_moving =
+    conditional_t<std::is_move_constructible<intrinsic_t<T>>::value,
+        // return an rvalue ref via T(std::move(value)) which triggers a move construction
+        intrinsic_t<T>,
+        // this doesn't work for abstract types: cast these to an rvalue reference
+        typename std::add_rvalue_reference<intrinsic_t<T>>::type>;
+
+// A plain argument type will use lvalue references (triggering copy semantics)
```

Comment:
By "plain", you're talking about "by value" arguments?

Reply:
Exactly.

Reply:
Fixed.